plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/LazyMaps.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import static com.google.common.base.Suppliers.memoize;
+
+public final class LazyMaps
+{
+    private LazyMaps() {}
+
+    public static <K, V> Map<K, V> of(Supplier<Set<Map.Entry<K, V>>> supplier)
+    {
+        Supplier<Set<Map.Entry<K, V>>> cachedSupplier = memoize(supplier::get);
+        return new AbstractMap<>()
```

Comment:
This is unlikely a performant implementation of a map.
Last time i implemented Map interface from scratch like this, i spend at least couple days writing the code to have it reasonable. (This was the first approach to have immutable identity-based collections for the planner, so early 2017 and my memory can be failing me)